src/components/select-field/BaseSelectField.js

```diff
@@ -480,7 +480,7 @@ class BaseSelectField extends React.Component<Props, State> {
 
             const isSelected = selectedValues.includes(value);
 
-            const isClearOption = shouldShowClearOption && value === 'clear' && index === 0;
+            const isClearOption = shouldShowClearOption && value === CLEAR;
```

Comment:
Using the magic constant CLEAR should be robust enough for us to not have to use index === 0.